packages/cfd/src/Containers/__tests__/cfd-top-up-demo-modal.spec.js

```diff
@@ -0,0 +1,163 @@
+        document.body.appendChild(modal_root_el);
+    });
+    afterAll(() => {
+        document.body.removeChild(modal_root_el);
+    });
+
+    it('should render <CFDTopUpDemoModal /> correctly', () => {
+        render(<CFDTopUpDemoModal {...mock_props} />);
+    });
+
+    it('should render the button texts correctly', () => {
+        render(<CFDTopUpDemoModal {...mock_props} />);
+        expect(screen.getByText('Fund top up')).toBeInTheDocument();
+        expect(screen.getByText('Current balance')).toBeInTheDocument();
+        expect(screen.getByText('Demo Financial account')).toBeInTheDocument();
+        expect(screen.queryByTestId('dt_top-up-virtual-description')).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: /Top up/i })).toBeInTheDocument();
+    });
+
+    it('should render the proper balance in the current balance', () => {
+        render(<CFDTopUpDemoModal {...mock_props} />);
+        expect(screen.getByText('700.00')).toBeInTheDocument();
+    });
+
+    it('should disable the top up button if the balance is higher than 1000 USD', () => {
+        render(
+            <CFDTopUpDemoModal
+                {...mock_props}
+                current_account={{ category: 'demo', type: 'financial', balance: 2000 }}
+            />
+        );
+        const top_up_btn = screen.getByRole('button', { name: /Top up/i });
+        expect(top_up_btn).toBeDisabled();
+    });
+
+    it('should enable the top up button if the balance is lower than 1000 USD', () => {
+        render(
+            <CFDTopUpDemoModal
+                {...mock_props}
+                current_account={{ category: 'demo', type: 'financial', balance: 500 }}
+            />
+        );
+        const top_up_btn = screen.getByRole('button', { name: /Top up/i });
+        expect(top_up_btn).toBeEnabled();
+    });
+
+    it('should render the success dialog component if the user has less than 1000 USD and clicks on top up', () => {
+        render(
+            <CFDTopUpDemoModal
+                {...mock_props}
+                current_account={{ category: 'demo', type: 'financial', balance: 500 }}
+            />
+        );
+        const top_up_btn = screen.getByRole('button', { name: /Top up/i });
+        fireEvent.click(top_up_btn);
+        expect(screen.getByText('Success Dialog')).toBeInTheDocument();
+    });
+
+    it('should render the success component if the is_top_up_virtual_success is true', () => {
+        render(
+            <CFDTopUpDemoModal
+                {...mock_props}
+                current_account={{ category: 'demo', type: 'financial', balance: 500 }}
+                is_top_up_virtual_success={true}
+                is_top_up_virtual_open={false}
+            />
+        );
+        expect(screen.getByText('Success Dialog')).toBeInTheDocument();
+        expect(screen.queryByTestId('dt_top-up-virtual-description')).not.toBeInTheDocument();
+        screen.debug();
+    });
+});
```

Comment:
can you please add the test coverage result to the redmine card

Reply:
suree